recipes/repo.rb

```diff
@@ -272,3 +272,15 @@
     action [:start, :enable]
   end
 end
+
+# Configure read-through container registry cache
+if node['ros_buildfarm']['repo']['container_registry_cache_enabled']
+  package 'docker-registry'
+  cookbook_file '/etc/docker/registry/config.yml' do
+    source 'docker-registry-config.yml'
+    notifies :restart, 'service[docker-registry]'
```

Comment:
Do this restart just before the service do anything different than just remove it?

Reply:
notification only occurs when the file is changed. So when chef is run and this file doesn't change, it won't restart the service. But if the file does change and the service is always running, without the notification to restart the docker-registry service wouldn't pick up the new config.

Reply:
thanks for the explanation.